File: tests/Search/BreadthFirstSearchTest.php
<?php

namespace GulDmitry\Algorithms\Tests;

use GulDmitry\Algorithms\Search\BreadthFirstSearch;
use PHPUnit\Framework\TestCase;

class BreadthFirstSearchTest extends TestCase
{
    public function testBFS()
    {
        $graph = [];
        $graph['you'] = ['alice', 'bob', 'claire'];
        $graph['bob'] = ['anuj', 'peggy'];
        $graph['alice'] = ['thom', 'jonny'];
        $graph['claire'] = ['you'];
        $graph['anuj'] = [];
        $graph['peggy'] = [];
        $graph['thom'] = [];
        $graph['jonny'] = [];

        $bfs = new BreadthFirstSearch();

        $this->assertEquals(
            'thom',
            $bfs->search($graph, 'you', function (string $edge): bool {
                return $edge === 'thom';
            })
        );
        $this->assertNull($bfs->search($graph, 'you', function (string $edge): bool {
            return $edge === 'mahmed';
        }));
    }
}

File: tests/go/Search/breadth-first_search_test.go
package Search

import "testing"
import (
	search "github.com/guldmitry/algorithms/Search"
)

func TestBFS(t *testing.T) {
	var graph = map[string][]string{
		"you": {
			"alice", "bob", "claire",
		},
		"bob": {
			"anuj", "peggy",
		},
		"alice": {
			"thom", "jonny",
		},
		"claire": {
			"you",
		},
		"anuj":  {},
		"peggy": {},
		"thom":  {},
		"jonny": {},
	}

	if "thom" != search.BFS(graph, "you", func(edge string) bool {
		return edge == "thom"
	}) {
		t.Error("Element was not found.")
	}

	if "" != search.BFS(graph, "you", func(edge string) bool {
		return edge == "mahmed"
	}) {
		t.Error("Element was not found.")
	}
}
